### DISCUSSION TITLE
How to access Fastly ClientInfo interface data from Hono request?

### DISCUSSION THREAD

**Discussion by alex8bitw** [honojs/hono#2030] [Q&A]
Hi,

 I'm trying to pull the client.address as per:  https://developer.fastly.com/solutions/examples/client-public-ip-api-at-the-edge/

I don't see a way to access the ClientInfo fastly interface.

Example:
https://fiddle.fastly.dev/fiddle/3b001f5d

References:
https://js-compute-reference-docs.edgecompute.app/docs/globals/FetchEvent/
https://github.com/fastly/js-compute-runtime/blob/3cd6844faa71194b3cac7d7dfcc366c3b2524242/types/globals.d.ts#L260-L266

**Comment by alex8bitw** [comment]
Well, I can now answer this myself; `c.executionCtx` can be used.  So for my request specifically, it would be:
` console.log("client.address:", c.executionCtx.client.address);`

**Reply by Swarz** [reply]
Error: This context has no ExecutionContext

 'executionCtx' does not always exist
